src/IIIFPresentation/API/Features/Storage/Requests/PostHierarchicalCollection.cs

```diff
@@ -0,0 +1,130 @@
+    string slug, 
+    UrlRoots urlRoots,
+    string rawRequestBody) : IRequest<ModifyEntityResult<Collection, ModifyCollectionType>>
+{
+    public int CustomerId { get; } = customerId;
+
+    public string Slug { get; } = slug;
+    
+    public UrlRoots UrlRoots { get; } = urlRoots;
+    
+    public string RawRequestBody { get; } = rawRequestBody;
+}
+
+public class PostHierarchicalCollectionHandler(
+    PresentationContext dbContext,    
+    ILogger<CreateCollection> logger,
+    IBucketWriter bucketWriter,
+    IIdGenerator idGenerator,
+    IOptions<ApiSettings> options)
+    : IRequestHandler<PostHierarchicalCollection, ModifyEntityResult<Collection, ModifyCollectionType>>
+{
+    private readonly ApiSettings settings = options.Value;
+    
+    public async Task<ModifyEntityResult<Collection, ModifyCollectionType>> Handle(PostHierarchicalCollection request,
+        CancellationToken cancellationToken)
+    {
+        var splitSlug = request.Slug.Split('/');
+
+        var parentSlug = String.Join(String.Empty, splitSlug.Take(..^1));
+        var parentCollection =
+            await dbContext.RetriveHierarchicalCollection(request.CustomerId, parentSlug, cancellationToken);
+        
+        if (parentCollection == null) return ErrorHelper.NullParentResponse<Collection>();
+
+        string id;
+        
+        try
+        {
+            id = await dbContext.Collections.GenerateUniqueIdAsync(request.CustomerId, idGenerator, cancellationToken);
+        }
+        catch (ConstraintException ex)
+        {
+            logger.LogError(ex, "An exception occured while generating a unique id");
+            return ErrorHelper.CannotGenerateUniqueId<Collection>();
+        }
+
+        Collection collectionFromBody;
+        try
+        {
+            collectionFromBody = request.RawRequestBody.FromJson<Collection>();
+            collectionFromBody.Id = $"{request.UrlRoots.BaseUrl}/{request.CustomerId}/{request.Slug}";
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Error attempting to validate collection is IIIF");
+            return ErrorHelper.CannotValidateIIIF<Collection>();
+        }
+
+        var thumbnails = collectionFromBody.Thumbnail?.Select(x => x as Image).ToList();     
+        
+        var dateCreated = DateTime.UtcNow;
+        var collection = new DatabaseCollection.Collection
+        {
+            Id = id,
+            Parent = parentCollection.Id,
+            Slug = splitSlug.Last(),
+            Created = dateCreated,
+            Modified = dateCreated,
+            CreatedBy = Authorizer.GetUser(),
+            CustomerId = request.CustomerId,
+            IsPublic = collectionFromBody.Behavior != null && collectionFromBody.Behavior.IsPublic(),
+            IsStorageCollection = false,
+            Label = collectionFromBody.Label,
+            Thumbnail = thumbnails!?.GetThumbnailPath()
+        };
+        
+        await using var transaction = 
```

Comment:
Is the transcation required here? `TrySaveCollection` calls `SaveChangesAsync` which will [be in a transaction](https://learn.microsoft.com/en-us/ef/core/saving/transactions#default-transaction-behavior) without you needing to do it externally. If `SaveChanges` fails, `saveErrors` will be returned. If `SaveChanges` works then the transaction will always be committed as `WriteToBucket` never throws